c7n/resources/sagemaker.py

```diff
@@ -107,6 +107,53 @@ def _augment(j):
             return list(filter(None, w.map(_augment, jobs)))
 
 
+@resources.register('sagemaker-transform-job')
+class SagemakerTransformJob(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'sagemaker'
+        enum_spec = ('list_transform_jobs', 'TransformJobSummaries', None)
+        detail_spec = (
+            'describe_transform_job', 'TransformJobName', 'TransformJobName', None)
+        id = 'TransformJobArn'
+        name = 'TransformJobName'
+        date = 'CreationTime'
+        dimension = None
+        filter_name = None
+
+    permissions = (
+        'sagemaker:ListTransformJobs', 'sagemaker:DescribeTransformJobs',
+        'sagemaker:ListTags')
+
+    def __init__(self, ctx, data):
+        super(SagemakerTransformJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'InProgress'}]))
+
+    def resources(self, query=None):
+        for q in self.queries:
+            if q is None:
+                continue
+            query = query or {}
+            for k, v in q.items():
+                query[k] = v
+        return super(SagemakerTransformJob, self).resources(query=query)
+
+    def augment(self, jobs):
+        client = local_session(self.session_factory).client('sagemaker')
+
+        def _augment(j):
+            tags = self.retry(client.list_tags,
+                ResourceArn=j['TransformJobArn'])['Tags']
+            j['Tags'] = tags
+            return j
+
+        jobs = super(SagemakerTransformJob, self).augment(jobs)
+        with self.executor_factory(max_workers=1) as w:
```

Comment:
for single threaded (max_workers=1) its faster/simpler to loop through the list.

Reply:
changed it use universal augment and two workers (based on implementation in sqs.py)